packages/gallery/src/components/gallery/proGallery/slideshowView.js

```diff
@@ -458,22 +458,37 @@ class SlideshowView extends GalleryComponent {
 
   handleSlideshowKeyPress(e) {
     e.stopPropagation();
-    const nextKeys = [32, 33, 34, 37, 38, 39, 40]
+    const relevantKeys = [32, 33, 34, 37, 38, 39, 40, 27];
     // key code -> 32=space, 37=left, 38=up, 39=right, 40=down
     // charCode -> , 33=page up, 34=page down
     const code = e.charCode || e.keyCode
 
-    if (nextKeys.includes(code)) {
+    if (relevantKeys.includes(code)) {
       e.preventDefault();
+      const activeItemIdx = window.document.activeElement.getAttribute('data-idx');
+
+      const shouldFocusOutOfViewComponent = activeItemIdx &&
+        (this.props.totalItemsCount - 1) === Number(activeItemIdx) &&
+        Number(activeItemIdx) === this.state.currentIdx;
+
+      switch (code) {
+        case 40: // down
+          if (shouldFocusOutOfViewComponent) {
```

Comment:
You can remove the long `switch` statement to a one liner 

```
if ((code == 27) || (code === 40 && shouldFocusOutOfViewComponent)) { 

utils.focusGalleryElement(this.props.galleryContainerRef) }
```

Reply:
is "utils.focusGalleryElement(this.props.**outOfViewComponent**)" on one condition and "utils.focusGalleryElement(this.props.**galleryContainerRef**)" in the other